### ISSUE TITLE
Generic ZigBee RGBW Light

### ISSUE THREAD

**Issue by autolog** [GledholtHall/beta-3#6]
It seems that the **color** property is missing from the list of properties for a Generic ZigBee RGBW Light.

This is a screenshot from MQTT Explorer:

<img width="465" alt="generic_zigbee_rgbw_light" src="https://user-images.githubusercontent.com/11138074/109293772-2d5a9900-7824-11eb-85f1-e173e4c938af.png">

**Comment by GledholtHall** [comment]
Yes, I'll add that - if you find missing or spurious $properties elsewhere do let me know.

**Comment by cesarvoggithub** [comment]
Yes, I have also found this issue. Because @autolog updated his Indigo Plugin substantially over the last 3 or 4 days, I can now turn on/turn off/set brightness level of color lamps (as well as actual dimmers), from either Indigo and/or Hubitat Elevation and have it immediately reflected on the other platform, but I can only set colors from Indigo. The issue @autolog raised above is probably what's causing what I'm seeing. With Jon's last changes on his plugin, I think 80% of my Indigo natively defined lights (either controled by switches or dimmers) are already controllable from either platform and the other platform correctly "picks up" the change. When color lamps properties are added to the MQTT App/Driver, I believe that 95% of my needs will be covered. Excellent work Kevin and Jon!!!

**Comment by GledholtHall** [comment]
I will add Color, it’s a mistake that it’s missing.  Not sure what and how Jon uses $properties but hope it helps.

RGB bulbs are a bit of a concern to me. I think I handle them fully & correctly from HE out to homie and the associated control but I don’t think discovery for RGB works back into HE. So you probably can’t create a HE virtual RGB device linked to a real Indigo one.    

That maybe the issue you’re finding Cesar where the HE device is virtual.  It should work fine when Indigo RGB device is virtual   It won’t be an about to happen feature either as I’m focussing on getting all the current aspects tidied for the beta 4 release.  You might be able to improve things by manually editing the associated color…/cmd topics.

**Comment by autolog** [comment]
I use $properties to determine the device type and what attributes it supports. As color was missing, I used the the other color settings to determine that it was a colour device. I will update my plugin when release pre25 is available.

At the moment I am not publishing any colour information for discovery but that is getting near the top of the list.

I haven't done any groovy programming (yet). :)

**Comment by GledholtHall** [comment]
I think you’re doing it absolutely right, although I feel homie as a structure definition is missing the easy ability to recreate a device rather than entities. not so bad with lights, switches etc but a problem for more complex ones.

quick Q .. Does Indigo support the Home Assistant MQTT Discovery protocol , either for import or export of devices ?  I’m guessing not.

**Comment by autolog** [comment]
>  quick Q .. Does Indigo support the Home Assistant MQTT Discovery protocol , either for import or export of devices ? I’m guessing not.

Outside the scope of this App but yes: https://www.indigodomo.com/pluginstore/233/

**Comment by autolog** [comment]
>  Outside the scope of this App but yes: https://www.indigodomo.com/pluginstore/233/

My mistake, I thought I was replying to Cesar, hence "out of scope" coment :) 

Also, I don't think that plugin uses MQTT.

**Comment by GledholtHall** [comment]
> I will add color $property, it’s an oversight that it’s missing. Not sure how Jon uses $properties but hope it helps.

As a temporary workaround you can force the `color `$property  to be included by enabling a color device in the Color Control Light Devices dropdown
